backend/src/controllers/userController.ts

```diff
@@ -1,24 +1,58 @@
-import User from "src/models/User";
+import User, { UserType } from "src/models/User";
+import { matchedData, validationResult } from "express-validator";
 import asyncHandler from "express-async-handler";
 import createHttpError from "http-errors";
+import validationErrorParser from "src/util/validationErrorParser";
+import Company from "src/models/Company";
+
+interface BaseUserResponse {
+  _id?: string;
+  email: string;
+  name: string;
+  type: string;
+}
+
+interface StudentResponse extends BaseUserResponse {
+  linkedIn?: string;
+  phoneNumber?: string;
+  major?: string;
+  classLevel?: string;
+}
+
+interface AlumniResponse extends BaseUserResponse {
+  linkedIn?: string;
+  phoneNumber?: string;
+  company?: string;
```

Comment:
I figure that an interface for CompanyResponse is not really necessary for now (at least at this stage it would be redundant if we're not returning different responses for different cases like as with Users with StudentResponses vs AlumniResponses).